### PR TITLE
Allow for monitoring job status in scripts

### PR DESCRIPTION
**Pull request by nonhermitian**
### Summary
In the tutorials there are a lot of places where the user must check the status of a job with a while-loop.  We have fixed this in notebooks, but the async calls used there are not beneficial for script users.  Here we add the `'monitor` arg to `job.status()` which will automatially check the status of a job in the same manner as the magic version does.  However, this version is blocking, and thus is suitable for scripts.

### Details and comments
```python
from qiskit import *
from qiskit.tools import job_monitor
from qiskit import IBMQ
device = IBMQ.get_backend('ibmqx4')

q = QuantumRegister(2)
c = ClassicalRegister(2)
qc= QuantumCircuit(q,c)
qc.h(q[0])
qc.cx(q[0],q[1])
qc.measure(q,c)
job = execute(qc, device)
job_monitor(job)
```

ezgif com-video-to-gif

ezgif com-video-to-gif 2

### CODE REVIEW

**Comment by jaygambetta** [comment]
I don't think we need this when the while loop works. Let's discuss this after the release.

**Comment by nonhermitian** [comment]
I am fine with that. It is more of an useability thing.

**Comment by jaygambetta** [comment]
paul can you paste the output and why do we need to use the std out.

**Comment by nonhermitian** [comment]
monitor

**Comment by jaygambetta** [comment]
Why can’t this be a function in wrapper than 

Job = execute 
JobMonitor(job) 
Result =job.result()

**Comment by nonhermitian** [comment]
It can be. Will move.

**Comment by jaygambetta** [comment]
Great I make an issue of how I think it should work.

File: qiskit/__init__.py [hunk 1]
```diff
@@ -44,7 +44,7 @@
 __path__ = pkgutil.extend_path(__path__, __name__)
 
 from .wrapper._wrapper import (compile, execute, load_qasm_string,
-                               load_qasm_file, least_busy, qobj_to_circuits)
+                               load_qasm_file, least_busy, qobj_to_circuits,)
```
Comment (jaygambetta) [inline qiskit/__init__.py]: i think you have a copy paste error here

File: qiskit/wrapper/_monitor.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    def _checker(job, interval):
+        status = job.status()
+        msg = status.value
+        prev_msg = msg
+        msg_len = len(msg)
+
+        sys.stdout.write('\r%s: %s' % ('Job Status', msg))
+        while status.name not in ['DONE', 'CANCELLED', 'ERROR']:
+            time.sleep(interval)
```
Comment (jaygambetta) [inline qiskit/wrapper/_monitor.py]: rather than use string can we import JobStatus here

**Comment by atilag** [comment]
I'm working on a Publisher/Subscriber implementation that can be used in this case, to avoid Jobs directly printing to the screen. As any other event-driven system, publishers publish events and subscribers react to the events they subscribe, so in this specific use-case, the Job will act as a publisher, emitting `job.status` events with the status as payload, and the monitor code located in Wrapper/Jupyter tools will subscribe to this event so they can do whatever they want with the status, like printing to screen.

File: qiskit/__init__.py (hunk 1 above)
Comment (nonhermitian) [reply qiskit/__init__.py]: Thanks for catching this

File: qiskit/wrapper/_monitor.py (hunk 2 above)
Comment (nonhermitian) [reply qiskit/wrapper/_monitor.py]: I am printing to the screen, so kind need a string.

**Comment by nonhermitian** [comment]
I still need to work on this.  I had to stop once all the other fun issues arose (from trying to test this PR).  Will fix shortly.

**Comment by nonhermitian** [comment]
Ok, this has been changed so that monitoring jobs uses ipywidgets (like the magic function does) in notebooks, and you can run async.  Scripts default to text based version that is blocking.

**Comment by jaygambetta** [comment]
what is the status of this. @atilag.

**Comment by atilag** [comment]
I have finished the Publisher/Subscriber, I'll PR next Monday (or maybe Tomorrow if I have time).
I have decoupled the ProgressBar from the Transpiler as it was the easier use-case, but once I submit the PR we can see the mechanism and discuss about how to apply in this scenario.

**Comment by 1ucian0** [comment]
On hold waiting for #1117 .

**Comment by jaygambetta** [comment]
lets not wait until #1117 as this is only for progress bar. We should merge this and we can use a pub/sub behind when it is all done. 

related to #993

**Comment by jaygambetta** [comment]
@nonhermitian can you fix the conflicts.

**Comment by nonhermitian** [comment]
The tests fail on Travis, but pass locally, so I am not sure what to do about it.  Also, here is what the `job_monitor` looks like in a script:

ezgif com-video-to-gif 2

**Comment by jaygambetta** [comment]
looks good both the

File: test/python/test_wrapper.py [hunk 3]
```diff
@@ -57,6 +57,19 @@ def test_qobj_to_circuits_with_qobj_no_qasm(self):
...
+        """Test job_monitor"""
+        qreg = QuantumRegister(2)
+        creg = ClassicalRegister(2)
+        qc = QuantumCircuit(qreg, creg)
+        qc.h(qreg[0])
+        qc.cx(qreg[0], qreg[1])
+        qc.measure(qreg, creg)
+        backend = Aer.get_backend('qasm_simulator')
+        job_sim = execute([qc]*10, backend)
+        qiskit.wrapper.job_monitor(job_sim)
```
Comment (jaygambetta) [inline test/python/test_wrapper.py]: lets move from wrapper to tools

**Comment by jaygambetta** [comment]
also @nonhermitian can you update the screen shot for the juypter as I think you are now using job_moniter as the function call.

**Comment by nonhermitian** [comment]
@jaygambetta Updated examples are in the original summary

**Comment by 1ucian0** [comment]
Onhold until #1117 is merged.

**Comment by jaygambetta** [comment]
@nonhermitian removing off hold. 

Could you fix the conflicts and then also lets make a tutorial on this

**Pull request merged**